packages/webapp/src/containers/Finances/TransactionFilter/index.jsx

```diff
@@ -38,7 +38,27 @@ const TransactionFilter = () => {
   const transactionsFilter = useSelector(transactionsFilterSelector);
   const dispatch = useDispatch();
 
+  const checkActiveFilters = (filterType) => {
+    // Check if filterType ("EXPENSE_TYPE" or "REVENUE_TYPE") has active filters:
+    const hasActiveFilters =
+      tempTransactionsFilter[filterType] &&
+      Object.values(tempTransactionsFilter[filterType]).some((filter) => {
+        return filter.active === true;
+      });
+
+    // If filterType has no active filters, reset its value to undefined:
+    if (!hasActiveFilters && tempTransactionsFilter[filterType]) {
+      setTempTransactionsFilter((prevState) => {
+        prevState[filterType] = undefined;
+        return prevState;
+      });
```

Comment:
Thank you for working on this!
Would it be a problem if we did this in `onChange` that sets `tempTransactionsFilter`? It would be nice if we could set the correct value in the first place!

Reply:
Nice suggestion. Thank you 🙂 

I just refactored the full function and moved its execution into the `onChange` event, leaving `handleApply` as it was before this PR.